action/rlp_tx.go

```diff
@@ -9,17 +9,18 @@ import (
 	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/iotexproject/go-pkgs/crypto"
 	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
 	"github.com/pkg/errors"
 	"golang.org/x/crypto/sha3"
 )
 
-func rlpRawHash(rawTx *types.Transaction, chainID uint32) (hash.Hash256, error) {
-	h := types.NewEIP155Signer(big.NewInt(int64(chainID))).Hash(rawTx)
+func rlpRawHash(rawTx *types.Transaction, signer types.Signer) (hash.Hash256, error) {
```

Comment:
signer and hasher are of different roles

Reply:
chatted offline, it's probably not worth the effort to define an interface `Hasher` for this, the code calling `signer.Hash()` is quite clear